### PR TITLE
Replaced EmailAddress class with regex for validating email addresses

### PR DESCRIPTION
**Pull request by click-alexvladut**
The newer HTML5 standard include a more loose validation of the email addresses than the library Cheddar uses at the moment. This validation class is replaced with a regular expression specified by W3C at: http://w3c.github.io/html/sec-forms.html#email-state-typeemail

### CODE REVIEW

File: commons/commons-lang/src/test/java/com/clicktravel/common/random/RandomsTest.java [hunk 1]
```diff
@@ -65,7 +69,11 @@ public void shouldReturnRandomEmailAddress() {
         for (int n = 0; n < SAMPLE_SIZE; n++) {
             final String randomEmailAddress = Randoms.randomEmailAddress();
             assertNotNull(randomEmailAddress);
-            assertTrue(EmailAddress.isValidMailbox(randomEmailAddress));
+            try {
+                Check.isValidEmail("email", randomEmailAddress);
+            } catch (final ValidationException e) {
+                fail("Invalid random email addredd generated: " + randomEmailAddress);
```
Comment (clicktravel-steffan) [inline commons/commons-lang/src/test/java/com/clicktravel/common/random/RandomsTest.java]: Typo, should be `address`

File: commons/commons-lang/src/test/java/com/clicktravel/common/validation/CheckTest.java [hunk 2]
```diff
@@ -219,38 +218,68 @@ public void shouldThrowValidationException_givenEmptyEmail() {
...
+        validEmails.add("user@localserver");
+        validEmails.add("john..doe@example.com");
 
         // When
-        ValidationException actualException = null;
-        try {
-            Check.isValidEmail(field, email);
-        } catch (final ValidationException e) {
-            actualException = e;
+        final List<String> wrongValidatedEmails = new ArrayList<>();
```
Comment (clicktravel-steffan) [inline commons/commons-lang/src/test/java/com/clicktravel/common/validation/CheckTest.java]: Rename this variable to `invalidatedEmails`

File: commons/commons-lang/src/test/java/com/clicktravel/common/validation/CheckTest.java [hunk 3]
```diff
@@ -219,38 +218,68 @@ public void shouldThrowValidationException_givenEmptyEmail() {
...
+        invalidEmails.add("this\\ still\\\"not\\allowed@example.com");
+        invalidEmails.add("john.doe@example..com");
 
         // When
-        ValidationException actualException = null;
-        try {
-            Check.isValidEmail(field, email);
-        } catch (final ValidationException e) {
-            actualException = e;
+        final List<String> wrongValidatedEmails = new ArrayList<>();
```
Comment (clicktravel-steffan) [inline commons/commons-lang/src/test/java/com/clicktravel/common/validation/CheckTest.java]: Rename this variable to `validatedEmails`

**Comment by click-alexvladut** [comment]
@clicktravel-steffan Addressed your comments. Please have another look.

**Comment by clicktravel-steffan** [comment]
@click-alexvladut All looks good now

**Pull request merged**